### ISSUE TITLE
Add server validation of documents

### ISSUE THREAD

**Issue by deontologician** [rethinkdb/horizon#138]
This is to enable a write to go through by validating that the resulting document is valid, but is separate from security rules

**Comment by josephg** [comment]
Yeah this would be great. Its irresponsible to allow clients to store arbitrary documents - you really want some (simple) schema validation for almost anything you store.

Calling schema validation code from inlined javascript strings in `config.toml` is super awkward.

**Comment by kjvalencik** [comment]
I agree that inlined js is very awkward. I think a better option would be a JS file to require. Otherwise, if you want syntax highlighting, to add unit tests, etc., then you need to template and build the toml, which is also awkward and feels very wrong.

**Comment by sachinbhutani** [comment]
It would be great, if it could be a generic hook, which may be used for validation change update or something else. 
Also, could it be isomorphic ?

**Comment by kiejo** [comment]
Making it isomorphic would be great as this should make it easier to implement offline support with optimistic updates.

What might also be interesting would be to include transformation capabilites which would allow using functions like `trim` or `toLowerCase` on fields.
